src/Sentry/Internal/UnsampledTransaction.cs

```diff
@@ -94,10 +94,10 @@ public override ISpan StartChild(string operation)
         return span;
     }
 
-    private class UnsampledSpan(UnsampledTransaction transaction) : NoOpSpan
+    public ISpan StartChild(string operation, SpanId spanId)
     {
-        public override ISpan StartChild(string operation) => transaction.StartChild(operation);
-
-        public override bool? IsSampled => false;
+        var span = new UnsampledSpan(this, spanId);
```

Comment:
question: do existing casts to `SpanTracer` require changes as well?

I did a quick search for casts to `SpanTracer`, and found these two usages:

https://github.com/getsentry/sentry-dotnet/blob/9beef00b325bd79c2cf43829f069142461aec113/src/Sentry.DiagnosticSource/Internal/DiagnosticSource/EFConnectionDiagnosticSourceHelper.cs#L58-L64

https://github.com/getsentry/sentry-dotnet/blob/9beef00b325bd79c2cf43829f069142461aec113/src/Sentry/SentrySpan.cs#L116-L129

I'm just wondering if they also need some changes?
I don't think so from what I can see ... but I just wanted to make sure.